lib/CXGN/BrAPI/v1/Images.pm

```diff
@@ -0,0 +1,466 @@
+    my $page = $self->page;
+    my $status = $self->status;
+    my $page_obj = CXGN::Page->new();
+    my $hostname = $page_obj->get_hostname();
+    my @data_files;
+
+    my $image_ids_arrayref = $params->{imageDbId} || ($params->{imageDbIds} || ());
+    my $image_names_arrayref = $params->{imageName} || ($params->{imageNames} || ());
+    my $stock_ids_arrayref = $params->{observationUnitDbId} || ($params->{observationUnitDbIds} || ());
+    my $phenotype_ids_arrayref = $params->{observationDbId} || ($params->{observationDbIds} || ());
+    my $descriptors_arrayref = $params->{descriptiveOntologyTerm} || ($params->{descriptiveOntologyTerms} || ());
+
+    my $image_search = CXGN::Image::Search->new({
+        bcs_schema=>$self->bcs_schema(),
+        people_schema=>$self->people_schema(),
+        phenome_schema=>$self->phenome_schema(),
+        image_name_list=>$image_names_arrayref,
+        description_list=>$descriptors_arrayref,
+        stock_id_list=>$stock_ids_arrayref,
+        image_id_list=>$image_ids_arrayref
+        # still need to implement in the search
+        # phenotype_id_list=>$phenotype_ids_arrayref,
+    });
+    my ($result, $total_count) = $image_search->search();
+
+    my @data;
+    foreach (@$result) {
+        my $image = SGN::Image->new($self->bcs_schema()->storage->dbh(), $_->{'image_id'});
+        my @cvterms = $image->get_cvterms();
+        my $url = $hostname . $image->get_image_url('medium');
+        my $filename = $image->get_filename();
+        my $size = (stat($filename))[7];
+        my ($width, $height) = imgsize($filename);
+
+        # Process cvterms
+        my @cvterm_names;
+        foreach (@cvterms) {
+            push(@cvterm_names, $_->name);
+        }
+
+        push @data, {
+            additionalInfo => {
+                observationLevel => $_->{'stock_type_name'},
+                observationUnitName => $_->{'stock_uniquename'},
+                tags =>  $_->{'tags_array'},
+            },
+            copyright => $_->{'image_username'} . " " . substr($_->{'image_modified_date'},0,4),
+            description => $_->{'image_description'},
+            descriptiveOntologyTerms => \@cvterm_names,
+            imageDbId => $_->{'image_id'},
+            imageFileName => $_->{'image_original_filename'},
+            imageFileSize => $size,
+            imageHeight => $height,
+            imageWidth => $width,
+            imageName => $_->{'image_name'},
+            imageTimeStamp => $_->{'image_modified_date'},
+            imageURL => $url,
+            mimeType => _get_mimetype($_->{'image_file_ext'}),
+            observationUnitDbId => $_->{'stock_id'},
+            # location and linked phenotypes are not yet available for images in the db
+            imageLocation => {
+                geometry => {
+                    coordinates => [],
+                    type=> '',
+                },
+                type => '',
+            },
+            observationDbIds => [],
```

Comment:
observations should be returned from the image search too, but only when phenotypes are actually linked to images
https://github.com/solgenomics/sgn/blob/master/lib/CXGN/Image/Search.pm#L475

Reply:
How are the observations linked to images? Specifically, what database tables are facilitating the connection? I see that you can associate variables to images through the metadata.md_image_cvterm table, but I haven't come across associating a phenotype with an   md_image object yet.

Reply:
They are linked by metadata.md_image -> phenome.nd_experiment_md_image -> nd_experiment -> nd_experiment_phenotype -> phenotype
This is still pretty new and is used when uploading a phenotype spreadsheet of type "associated images" or when doing phenotype extraction from drone imagery

Reply:
I updated the code so that if the user passes observationDbIds in with the POST/PUT images endpoint, the phenotypes passed are attached to the image. Also, the observationDbIds are returned in the image search and other image endpoints responses. 

To test, I did manual testing for expected results, and also confirmed with the new Field Book BrAPI syncing that the images endpoints in breedbase still work.

Reply:
thanks @ctucker3 👍